packages/grid/x-data-grid-premium/src/hooks/features/export/useGridExcelExport.tsx

```diff
@@ -53,21 +68,99 @@ export const useGridExcelExport = (apiRef: React.MutableRefObject<GridPrivateApi
   );
 
   const exportDataAsExcel = React.useCallback<GridExcelExportApi['exportDataAsExcel']>(
-    async (options) => {
-      logger.debug(`Export data as excel`);
+    async (options = {}) => {
+      const {
+        worker: workerFn,
+        exceljsPostProcess,
+        exceljsPreProcess,
+        columnsStyles,
+        includeHeaders,
+        getRowsToExport = defaultGetRowsToExport,
+        valueOptionsSheetName = 'Options',
+        ...cloneableOptions
+      } = options;
+
+      const sendExcelToUser = (buffer: any) => {
+        const blob = new Blob([buffer], {
+          type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+        });
+        exportAs<GridExportExtension>(blob, 'xlsx', options?.fileName);
+      };
 
-      const workbook = await getDataAsExcel(options);
-      if (workbook === null) {
+      if (!workerFn) {
+        apiRef.current.publishEvent('excelExportStateChange', true);
+
+        const workbook = await getDataAsExcel(options);
+        if (workbook === null) {
+          return;
+        }
+
+        const content = await workbook.xlsx.writeBuffer();
+        apiRef.current.publishEvent('excelExportStateChange', false);
+        sendExcelToUser(content);
         return;
       }
-      const content = await workbook.xlsx.writeBuffer();
-      const blob = new Blob([content], {
-        type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
-      });
 
-      exportAs<GridExportExtension>(blob, 'xlsx', options?.fileName);
+      if (exceljsPostProcess) {
```

Comment:
Same for the second if ?
```suggestion
      if (exceljsPostProcess && process.env !== "production") {
```